### PR TITLE
MVP support for themes

### PR DESCRIPTION
**Pull request by jschr**
https://app.asana.com/0/529054306396887/661328846979617

Depends on https://github.com/mirainc/mira-elements/pull/90

### CODE REVIEW

File: packages/mira-kit/src/prop-types/extractProperties.js [hunk 1]
```diff
@@ -47,6 +49,18 @@ export default function extractProperties(
       prop.properties = itemProperties;
     }
 
+    // TODO: validate that only one prop can exist with injectThemes.
```
Comment (KiaFathi) [inline packages/mira-kit/src/prop-types/extractProperties.js]: Is this happening below already?

File: packages/mira-kit/src/prop-types/ThemeType.js [hunk 2]
```diff
@@ -0,0 +1,12 @@
+import BaseType from './BaseType';
+
+export default class StringType extends BaseType {
```
Comment (KiaFathi) [inline packages/mira-kit/src/prop-types/ThemeType.js]: Should this be StringType or ThemeType?

File: packages/mira-kit/src/prop-types/ThemeType.js [hunk 3]
```diff
@@ -0,0 +1,12 @@
...
+  constructor(label = 'Theme') {
+    super(label, 'selection');
+    // Tells the consumer (dashboard or simulator) to inject
+    // user themes into this prop.
+    this.propType.injectThemes = true;
```
Comment (KiaFathi) [inline packages/mira-kit/src/prop-types/ThemeType.js]: Is this necessary here? Should we instead fork logic if the label is theme?

File: packages/mira-kit/src/prop-types/extractProperties.js (hunk 1 above)
Comment (jschr) [reply packages/mira-kit/src/prop-types/extractProperties.js]: Yep, will remove

File: packages/mira-kit/src/prop-types/ThemeType.js (hunk 2 above)
Comment (jschr) [reply packages/mira-kit/src/prop-types/ThemeType.js]: Yeah, typo!

File: packages/mira-kit/src/prop-types/ThemeType.js (hunk 3 above)
Comment (jschr) [reply packages/mira-kit/src/prop-types/ThemeType.js]: Just chatted with Aric, we're going with a proper `theme` type instead of having `injectThemes`

Comment (KiaFathi) [reply packages/mira-kit/src/prop-types/ThemeType.js]: Awesome, I like that more :)

**Comment by jschr** [comment]
Closing. Will be updating the screenshot service instead,

**Pull request merged**